service/main.go

```diff
@@ -506,3 +506,54 @@ func (srv *Service) listen() {
 		}
 	}
 }
+
+// ListenFor opens a tmp subscription to a topic and passes results to handler
+func (srv *Service) ListenFor(topic string, handler func(pid peer.ID, env *pb.Envelope) (*pb.Envelope, error)) context.CancelFunc {
```

Comment:
Service layer provides this helper for spinning up a tmp subscription at a random ID (the search request ID in the case of contact search)